lib/verbalize/action.rb

```diff
@@ -17,20 +17,24 @@ def self.included(target)
 
     module ClassMethods
       def input(*required_keywords, optional: [])
-        @inputs = required_keywords
+        @required_inputs = required_keywords
         @optional_inputs = optional
 
         class_eval Build.call(required_keywords, Array(optional))
       end
 
-      def inputs
-        @inputs || []
+      def required_inputs
+        @required_inputs || []
       end
 
       def optional_inputs
         @optional_inputs || []
       end
 
+      def inputs
+        required_inputs + optional_inputs
```

Comment:
we have two use-cases in master, both of which want both required and optional inputs to slice from a larger hash.